pandas/core/base.py

```diff
@@ -990,13 +991,73 @@ def factorize(self, sort=False, na_sentinel=-1):
         from pandas.core.algorithms import factorize
         return factorize(self, sort=sort, na_sentinel=na_sentinel)
 
-    def searchsorted(self, key, side='left'):
-        """ np.ndarray searchsorted compat """
+    _shared_docs['searchsorted'] = (
+        """Find indices where elements should be inserted to maintain order.
 
-        # FIXME in GH7447
-        # needs coercion on the key (DatetimeIndex does alreay)
-        # needs tests/doc-string
-        return self.values.searchsorted(key, side=side)
+        Find the indices into a sorted %(klass)s `self` such that, if the
+        corresponding elements in `v` were inserted before the indices, the
+        order of `self` would be preserved.
+
+        Parameters
+        ----------
+        v : array_like
```

Comment:
@jorisvandenbossche : I think you may have hit a Python bug in fact.
~~~.{Python}
>>> class Demo(object):
             def illustrate(self, a, b=2, c=3):
                   print("See, I worked, and 'a' is not a keyword argument.")
>>> demo = Demo()
>>> demo.illustrate(a=5)
"See, I worked, and 'a' is not a keyword argument."
~~~
This is a Python bug, which unfortunately we will have to live with, so yes, `values` should become `v` again...